examples/direct_fidelity_estimation.py

```diff
@@ -0,0 +1,82 @@
+"""Implements direct fidelity estimation.
+
+Direct Fidelity Estimation from Few Pauli Measurements
+https://arxiv.org/abs/1104.4695
+
+Practical characterization of quantum devices without tomography
+https://arxiv.org/abs/1104.3835
+"""
+
+import cirq
+import heapq
+import itertools
+import math
+import numpy
+
+
+def build_circuit():
+  # Builds an arbitrary circuit to test. The circuit is non Clifford to show the
+  # use of simulators.
+  qubits = cirq.LineQubit.range(3)
+  circuit = cirq.Circuit(
+      cirq.Z(qubits[0])**0.25,  # T-Gate, non Clifford.
+      cirq.X(qubits[1])**0.123,
+      cirq.X(qubits[2])**0.456)
+  return circuit, qubits
+
+
+def compute_characteristic_function(circuit, P_i, qubits, noise):
+  n = len(P_i)
+  d = 2**n
+
+  trace = 0  # rho_i / sigma_i in https://arxiv.org/pdf/1104.3835.pdf
+  for x in range(d):
+    xbin = numpy.binary_repr(x, width=n)
+
+    pauli_string = cirq.PauliString(dict(zip(qubits, P_i)))
+    display = cirq.approx_pauli_string_expectation(pauli_string, num_samples=1)
```

Comment:
I think this function has been removed from cirq `master`. See my comment on this function about how you might consider doing it instead